packages/x-date-pickers/src/YearPicker/YearPicker.tsx

```diff
@@ -21,6 +23,26 @@ const useUtilityClasses = (ownerState: any) => {
   return composeClasses(slots, getYearPickerUtilityClass, classes);
 };
 
+function useYearPickerDefaultizedProps<TDate>(
+  props: YearPickerProps<TDate>,
+  name: string,
+): DefaultizedProps<YearPickerProps<TDate>, 'minDate' | 'maxDate'> {
+  const utils = useUtils<TDate>();
+  const defaultDates = useDefaultDates<TDate>();
+  const themeProps = useThemeProps({
+    props,
+    name,
+  });
+
+  return {
+    ...themeProps,
+    minDate: parseNonNullablePickerDate(utils, themeProps.minDate, defaultDates.minDate),
+    maxDate: parseNonNullablePickerDate(utils, themeProps.maxDate, defaultDates.maxDate),
+    disablePast: false,
+    disableFuture: false,
```

Comment:
Such that theme  Props can override the `disableXxx`
```suggestion
  return {
    disablePast: false,
    disableFuture: false,
    ...themeProps,
    minDate: parseNonNullablePickerDate(utils, themeProps.minDate, defaultDates.minDate),
    maxDate: parseNonNullablePickerDate(utils, themeProps.maxDate, defaultDates.maxDate),
```